lib/signed-token-request.js

```diff
@@ -0,0 +1,161 @@
+"use strict";
+
+
+/**
+ * @typedef {Object} SignedTokenRequest
+ * @function download
+ * @function downloadToFile
+ * @function upload
+ * @function uploadFromFile
+ */
+
+
+/**
+ * Creates SignedTokenRequest class.
+ *
+ * @param {bluebird} bluebird
+ * @param {fs} fs
+ * @param {string} host
+ * @param {logger} logger
+ * @param {request} request
+ * @param {request} requestAsync
+ * @param {requestHandler} requestHandler
+ * @param {signer} signer
+ * @param {stream~PassThrough} StreamPassThrough
+ * @param {tokenDownloader} tokenDownloader
+ * @param {uri-js} Uri
+ * @return {SignedTokenRequest}
+ */
+module.exports = (bluebird, fs, host, logger, request, requestAsync, requestHandler, signer, StreamPassThrough, tokenDownloader, Uri) => {
+    var Bluebird;
+
+    Bluebird = bluebird;
+
+
+    /**
+     * Interface for signed requests.
+     */
+    class SignedTokenRequest {
+        /**
+         * Constructor for SignedTokenRequest.
+         *
+         * @param {string} accountId - Account ID.
+         * @param {boolean} isPublic
+         * @param {string} code
+         * @param {string} secret
+         */
+        constructor(accountId, isPublic, code, secret) {
+            this.accountId = accountId;
+            this.isPublic = isPublic;
+            this.code = code;
+            this.secret = secret;
+        }
+
+
+        /**
+         * Downloads content from OpenToken.
+         *
+         * @param {string} token
+         * @return {Promise.<string>}
+         */
+        download(token) {
```

Comment:
Style guide says that things returning promises should have `Async` as a suffix.  `downloadAsync()`, `uploadAsync()`.